clippy_lints/src/bool_to_int_with_if.rs

```diff
@@ -52,9 +53,9 @@ impl<'tcx> LateLintPass<'tcx> for BoolToIntWithIf {
 }
 
 fn check_if_else<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx rustc_hir::Expr<'tcx>) {
-    if let ExprKind::If(check, then, Some(else_)) = expr.kind
+    if let Some(If { cond, then, r#else: Some(r#else) }) = If::hir(expr)
```

Comment:
I'm considering if it would be nicer to suggest `i32::from(matches!(X, Y))` instead of ignoring `if let` cases completely. That would be shorter in most cases, but maybe not more readable. This is just a thought, so I'm fine with this as well.

Reply:
Not sure on that one, I think it's a bit too complex personally

Reply:
Okay, then let's just ignore this case. Creating the suggestion would also not be too much fun